www: rewriting "data fetching" code example

Asqit opened the pull request:
Creating pull request on issue #1553

Asqit committed:
www: rewriting "data fetching" code example

Creating pull request on issue #1553

iuioiua commented on docs/latest/getting-started/fetching-data.md:
```diff
@@ -9,56 +9,53 @@ during rendering. In real projects, this is often different. In many cases you
+These operations are all asynchronous. However, rendering is always synchronous.
+So, to fetch data dynamically, we can use our brand new `async-route`
+components, which abstract the old way of fetching data by writing the
+`Handlers`.
```
This entire section may not be needed now.

iuioiua commented on docs/latest/getting-started/fetching-data.md:
````diff
@@ -9,56 +9,53 @@ during rendering. In real projects, this is often different. In many cases you
 
 ```tsx
-// routes/github/[username].tsx
+// File: 'routes/github/[username].tsx'
````
```suggestion
// routes/github/[username].tsx
```
I think it's obvious enough that the commented path refers to the file in question. It's ok to leave the `File:` prefix out.

iuioiua commented on docs/latest/getting-started/fetching-data.md:
```diff
@@ -9,56 +9,53 @@ during rendering. In real projects, this is often different. In many cases you
+create a way to fetch the data, for example with the `fetch` API. The `req`
+parameter is mandatory, and it indicates that we will fetch some data. Then we
+need the `ctx` prop to get our route parameters (in this case, it's the
+username).
```
This page is not a guide on how to refactor the old handlers to new async route components. Can you rewrite this so it doesn't sound like one?

Asqit replied:
Thank you! Does this sound better? 

"
In order to fetch data, the route component must be asynchronous. The `req`
parameter is mandatory, and it indicates that certain data will be fetched. The
second `req` parameter is used to get the route parameters, such as the username
is this case.
"

iuioiua commented on docs/latest/getting-started/fetching-data.md:
```diff
@@ -9,56 +9,53 @@ during rendering. In real projects, this is often different. In many cases you
-    return ctx.render(user);
-  },
-};
+export default async function Page(req: Request, ctx: RouteContext) {
```
```suggestion
export default async function Page(_req: Request, ctx: RouteContext) {
```
As the `Request` object isn't being used, we prepend it with an `_`.

iuioiua reviewed (changes requested):
Great work! Just needs a bit of tweaking.

Asqit committed:
Update docs/latest/getting-started/fetching-data.md

Adding underscore in front of `req` parameter to indicate, that it is not used

Co-authored-by: Asher Gomez <ashersaupingomez@gmail.com>

Asqit committed:
Update docs/latest/getting-started/fetching-data.md

Fixing the comment indicating file location

Co-authored-by: Asher Gomez <ashersaupingomez@gmail.com>

Asqit committed:
Update fetching-data.md

Rewriting text, so that it does not sound like tutorial

Asqit committed:
Merge branch 'main' into main

iuioiua committed:
tweaks

iuioiua reviewed (approved):
I've made a few tweaks. Now, LGTM! Thanks for helping out, @Asqit.

Asqit committed:
Merge branch 'main' into main
